### PR TITLE
fix: make SET system type case-insensitive

### LINKED ISSUE #2701: lowercase sql mode values are not supported
When playing [MySQL Shell](https://dev.mysql.com/doc/mysql-shell/8.4/en/) with this library (e.g., loading a MySQL dump), one of the obstacles I encountered is that `mysqlsh` sends lowercase `SET sql_mode` statement, which is not supported by `go-mysql-server`. This issue can be reproduced with the built-in example:

```sh
cd _example
go run .
```
```sql
mysql -h127.0.0.1 -uroot -P3306;
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 1
Server version: 8.0.33 Dolt

Copyright (c) 2000, 2022, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> set sql_mode = 'no_auto_value_on_zero';
ERROR 1105 (HY000): value no_auto_value_on_zero was not found in the set

mysql> set sql_mode = 'NO_AUTO_VALUE_ON_ZERO';
Query OK, 1 row affected (0.00 sec)
```

### PR DESCRIPTION
**Pull request by fanyang01**
Resolves #2701

### CODE REVIEW

**Review by jycor — CHANGES_REQUESTED** [review]
Thanks for the contribution!
MySQL does use the `utf8mb4_0900_ai_ci` collation for `@@sql_mode`, so we should change that, but only that variable and not all system set type

File: sql/types/system_set.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ var _ sql.CollationCoercible = systemSetType{}
 
 // NewSystemSetType returns a new systemSetType.
 func NewSystemSetType(varName string, values ...string) sql.SystemVariableType {
-	return systemSetType{MustCreateSetType(values, sql.Collation_Default), varName}
+	return systemSetType{MustCreateSetType(values, sql.Collation_ascii_general_ci), varName}
```
[the remark this answers is not in the document]
Comment (fanyang01) [reply sql/types/system_set.go:38]: Hi @jycor, thank you for your review! I'm uncertain about which collation should be used for other system variables. Could you please direct me to where I can find MySQL's implementation?

```sql
mysql> set global log_output = 'table,file';
Query OK, 0 rows affected (0.00 sec)

mysql> select @@global.log_output;
+---------------------+
| @@global.log_output |
+---------------------+
| FILE,TABLE          |
+---------------------+
1 row in set (0.00 sec)

mysql> set global log_output = default;
Query OK, 0 rows affected (0.00 sec)

mysql> select @@global.log_output;
+---------------------+
| @@global.log_output |
+---------------------+
| FILE                |
+---------------------+
1 row in set (0.00 sec)

mysql> set global protocol_compression_algorithms = 'zLIB,Zstd';
Query OK, 0 rows affected (0.01 sec)

mysql> select @@global.protocol_compression_algorithms;
+------------------------------------------+
| @@global.protocol_compression_algorithms |
+------------------------------------------+
| zLIB,Zstd                                |
+------------------------------------------+
1 row in set (0.01 sec)

mysql> set global protocol_compression_algorithms = default;
Query OK, 0 rows affected (0.00 sec)

mysql> select @@global.protocol_compression_algorithms;
+------------------------------------------+
| @@global.protocol_compression_algorithms |
+------------------------------------------+
| zlib,zstd,uncompressed                   |
+------------------------------------------+
1 row in set (0.00 sec)
```

[the remark this answers is not in the document]
Comment (fanyang01) [reply sql/types/system_set.go]: I have learned the following facts from MySQL's source code:
- `protocol_compression_algorithms` is [defined to be a `Sys_var_charptr`](https://github.com/mysql/mysql-server/blob/61a3a1d8ef15512396b4c2af46e922a19bf2b174/sql/sys_vars.cc#L7296) and is [validated](https://github.com/mysql/mysql-server/blob/61a3a1d8ef15512396b4c2af46e922a19bf2b174/sql-common/compression.cc#L40) using the case-insensitive collation of `latin1`.
- `sql_mode` and `log_output` are defined as [`Sys_var_set`](https://github.com/mysql/mysql-server/blob/61a3a1d8ef15512396b4c2af46e922a19bf2b174/sql/sys_vars.h#L1624) and is [parsed](https://github.com/mysql/mysql-server/blob/61a3a1d8ef15512396b4c2af46e922a19bf2b174/sql/strfunc.cc#L88) using [the `to_upper` function of `system_charset_info`](https://github.com/mysql/mysql-server/blob/61a3a1d8ef15512396b4c2af46e922a19bf2b174/sql/strfunc.cc#L127).

**Review by jycor — APPROVED** [review]
Thanks for the changes and even digging into the MySQL source code.
LGTM!

**Pull request merged**